coral/src/app/features/topics/acl-request/forms/TopicConsumerForm.tsx

```diff
@@ -70,7 +71,27 @@ const TopicConsumerForm = ({
   }, [aclIpPrincipleType]);
 
   const { mutate, isLoading, isError, error } = useMutation({
-    mutationFn: createAclRequest,
+    mutationFn: async (payload: TopicConsumerFormSchema) => {
+      const { teamId, error } = await getTopicTeam({
+        topicName: payload.topicname,
+        patternType: payload.aclPatternType,
+      });
+
+      // teamId will actually never be undefined, but will be 0 when there is an error
+      // However, the openapi.yaml definition doesn't reflect that, so we need to check for undefined
+      // So that teamId in the happy path can be typed properly
+      if (error !== undefined || teamId === undefined) {
+        const errorMessage =
+          error || "There was an error fetching the topic team.";
+        // We need to throw an error here instead of setting an error on the field for two reasons:
+        // - we need to prevent the useMutatino to reach onSuccess (which would navigate away)
```

Comment:
```suggestion
        // - we need to prevent the useMutation to reach onSuccess (which would navigate away)
```

(I love `Mutatino` tho 🥺 )

Reply:
The cutest pokemon ^^

Reply:
[34240cb](https://github.com/Aiven-Open/klaw/pull/1702/commits/34240cbf6fe24193b8bed8156ab71e7d0b8de8ba)